assets/src/edit-story/components/library/panes/media/local/mediaPane.js

```diff
@@ -265,15 +276,32 @@ function MediaPane(props) {
               options={FILTERS}
               placement={Placement.BOTTOM_START}
             />
-            <Primary onClick={openMediaPicker}>
-              {__('Upload', 'web-stories')}
-            </Primary>
+            {isSearching && Boolean(media.length) ? (
+              <SearchCount>
+                {sprintf(
+                  /* translators: %d: number of results. */
+                  _n(
+                    '%d result found',
+                    '%d results found',
+                    media.length,
+                    'web-stories'
+                  ),
+                  media.length
+                )}
+              </SearchCount>
+            ) : (
+              <Primary onClick={openMediaPicker}>
```

Comment:
Upload button also needs to be hidden when `isSearching` is true, but even if there are no results.

Reply:
So maybe 

```
{isSearching && Boolean(media.length) && (<SearchCount>...)}
{!isSearching && <Primary...>}
```

Reply:
Good catch, updated!